acceptance-test/src/acceptance-test/java/org/zalando/nakadi/repository/db/SubscriptionDbRepositoryTest.java

```diff
@@ -219,4 +223,154 @@ private void checkSubscriptionCreatedFromSubscriptionBase(final Subscription sub
+                emptySet(), owiningApp, Optional.empty(), lastPageToken, 3);
+        assertThat(lastPage1.getNext(), nullValue());
+        assertThat(lastPage1.getPrev(), notNullValue());
+        assertThat(lastPage1.getItems(), equalTo(testSubscriptions.subList(6, 7)));
+        final SubscriptionDbRepository.ListResult lastPage2 = repository.listSubscriptions(
+                emptySet(), owiningApp, Optional.empty(), lastPageToken, 1);
+        assertThat(lastPage2, equalTo(lastPage1));
+    }
+
+    private List<Subscription> createTestSubscriptions(final String owiningApp, final int count) {
+        final List<Subscription> testSubscriptions = IntStream.range(0, count)
+                .mapToObj(i -> RandomSubscriptionBuilder.builder()
+                        .withOwningApplication(owiningApp).withEventType("et1").build())
+                .collect(toList());
+        testSubscriptions.forEach(this::insertSubscriptionToDB);
+        Collections.sort(testSubscriptions, Comparator.comparing(Subscription::getId));
+        return testSubscriptions;
+    }
+
+    @Test
+    public void testListing() {
+        // The test is a copy of SubscriptionDbRepositoryTest, that is to be deleted when deprecated method removed
+        final String owningApp = TestUtils.randomUUID();
+        final String owningApp2 = TestUtils.randomUUID();
+
+        final List<Subscription> testSubscriptions = ImmutableList.of(
+                RandomSubscriptionBuilder.builder().withOwningApplication(owningApp).withEventType("et1").build(),
+                RandomSubscriptionBuilder.builder().withOwningApplication(owningApp)
+                        .withEventTypes(ImmutableSet.of("et2", "et1")).build(),
+                RandomSubscriptionBuilder.builder().withOwningApplication(owningApp).withEventType("et1").build(),
+                RandomSubscriptionBuilder.builder().withOwningApplication(owningApp).withEventType("et2").build(),
+                RandomSubscriptionBuilder.builder().withOwningApplication(owningApp)
+                        .withEventTypes(ImmutableSet.of("et2", "et3")).build(),
+                RandomSubscriptionBuilder.builder().withOwningApplication(owningApp2).withEventType("et1").build(),
+                RandomSubscriptionBuilder.builder().withOwningApplication(owningApp2).withEventType("et2").build());
+        testSubscriptions.forEach(this::insertSubscriptionToDB);
+
+        final List<Subscription> expectedSubscriptions = testSubscriptions.stream()
+                .filter(sub -> owningApp.equals(sub.getOwningApplication()) && sub.getEventTypes().contains("et1"))
+                .sorted(Comparator.comparing(Subscription::getId))
+                .collect(toList());
+
+        final List<Subscription> subscriptions = repository.listSubscriptions(
+                ImmutableSet.of("et1"), Optional.of(owningApp), Optional.empty(), null, 10).getItems();
+        assertThat(subscriptions, equalTo(expectedSubscriptions));
+    }
```

Comment:
What do you think, does it make sense to use table-driven testing for tests of this class?

Reply:
I don't want to change anything in these tests as long as they work.  I just moved them here from the token lister class, unchanged.